src/esp/gfx/PbrShader.h

```diff
@@ -0,0 +1,442 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_GFX_PBRSHADER_H_
+#define ESP_GFX_PBRSHADER_H_
+
+#include <initializer_list>
+#include <vector>
+
+#include <Corrade/Containers/ArrayView.h>
+#include <Corrade/Containers/EnumSet.h>
+#include <Magnum/GL/AbstractShaderProgram.h>
+#include <Magnum/Math/Matrix4.h>
+#include <Magnum/Shaders/Generic.h>
+
+#include "esp/core/esp.h"
+
+namespace esp {
+
+namespace gfx {
+
+class PbrShader : public Magnum::GL::AbstractShaderProgram {
+ public:
+  // ==== Attribute definitions ====
+  /**
+   * @brief vertex positions
+   */
+  typedef Magnum::Shaders::Generic3D::Position Position;
+
+  /**
+   * @brief normal direction
+   */
+  typedef Magnum::Shaders::Generic3D::Normal Normal;
+
+  /**
+   * @brief 2D texture coordinates
+   *
+   * Used only if at least one of
+   * @ref Flag::BaseColorTexture, @ref Flag::NormalTexture and
+   * @ref Flag::RoughnessTexture @ref Flag::MetallicTexture is set.
+   */
+  typedef Magnum::Shaders::Generic3D::TextureCoordinates TextureCoordinates;
+
+  /**
+   * @brief Tangent direction with the fourth component indicating the handness.
+   *
+   * T = Tangent, B = BiTangent, N = Normal
+   *
+   * 1.0 means T, B, N form a right-handed coordinate;
+   * -1.0 means T, B, N form a left-handed coordinate;
+   *
+   * Used only if @ref Flag::NormalTexture is set.
+   */
+  typedef Magnum::Shaders::Generic3D::Tangent4 Tangent4;
+
+  enum : Magnum::UnsignedInt {
+    /**
+     * Color shader output. @ref shaders-generic "Generic output",
+     * present always. Expects three- or four-component floating-point
+     * or normalized buffer attachment.
+     */
+    ColorOutput = 0,
```

Comment:
While we're at it, this could be a Generic value as well:

```suggestion
    ColorOutput = Magnum::Shaders::Generic3D::ColorOutput,
```

Same for `ObjectIdOutput` below.

Reply:
OK.